### PR TITLE
passing getlogger to errorencoder on ancla level

### PR DESCRIPTION
**Pull request by mtrinh11**

### CODE REVIEW

**Comment by mtrinh11** [comment]
fixes https://github.com/xmidt-org/ancla/issues/89

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- Bumped ancla to v0.3.3 and changed server log source address field. [#231](https://github.com/xmidt-org/tr1d1um/pull/231)
+- Bumped ancla to v0.3.5, bumped argus to v0.6.0, and changed errorEncoder to pull logger from context. [#233](https://github.com/xmidt-org/tr1d1um/pull/233)
```
Comment (kristinapathak) [inline CHANGELOG.md:9]: From a release perspective, we don't need to know about intermediate increases in package versions.  I would restructure these lines like this:
```suggestion
- Bumped argus to v0.6.0. [#233](https://github.com/xmidt-org/tr1d1um/pull/233)
- Bumped ancla to v0.3.5:
  - Changed server log source address field. [#231](https://github.com/xmidt-org/tr1d1um/pull/231)
  - Changed errorEncoder to pull logger from context. [#233](https://github.com/xmidt-org/tr1d1um/pull/233)
```

File: common/utils.go [hunk 2]
```diff
@@ -115,10 +117,17 @@ func ForwardHeadersByPrefix(p string, from http.Header, to http.Header) {
 
 // ErrorLogEncoder decorates the errorEncoder in such a way that
 // errors are logged with their corresponding unique request identifier
-func ErrorLogEncoder(logger kitlog.Logger, ee kithttp.ErrorEncoder) kithttp.ErrorEncoder {
-	var errorLogger = logging.Error(logger)
+/*pass get logger func to pull logger out and use in the func returned; basically like the other ones */
```
Comment (kristinapathak) [inline common/utils.go:120]: This will be part of the go documentation.  Is the plan to remove this comment?

File: common/utils.go [hunk 3]
```diff
@@ -115,10 +117,17 @@ func ForwardHeadersByPrefix(p string, from http.Header, to http.Header) {
...
+	if getLogger == nil {
+		getLogger = func(_ context.Context) kitlog.Logger {
+			return nil
+		}
+	}
+
 	return func(ctx context.Context, e error, w http.ResponseWriter) {
-		errorLogger.Log(logging.ErrorKey(), e.Error(), "tid", ctx.Value(ContextKeyRequestTID).(string))
+		logger := getLogger(ctx)
+		logger.Log(logging.ErrorKey(), e.Error(), "tid", ctx.Value(ContextKeyRequestTID).(string))
```
Comment (kristinapathak) [inline common/utils.go:130]: We should check that the logger isn't `nil` before using it.

Comment (kristinapathak) [reply common/utils.go:130]: This log is also missing a message and a level.

File: common/utils_test.go [hunk 4]
```diff
@@ -82,7 +82,7 @@ func TestErrorLogEncoder(t *testing.T) {
 	e := func(ctx context.Context, _ error, _ http.ResponseWriter) {
 		assert.EqualValues("tid00", ctx.Value(ContextKeyRequestTID))
 	}
-	le := ErrorLogEncoder(logging.DefaultLogger(), e)
+	le := ErrorLogEncoder(GetLogger, e)
```
Comment (kristinapathak) [inline common/utils_test.go:85]: Can you add a test case where we pass `nil` for the logger?

**Review by kristinapathak — CHANGES_REQUESTED** [review]
Looks good! Left a few suggestions and questions.

File: CHANGELOG.md (hunk 1 above)
Comment (kristinapathak) [reply CHANGELOG.md:9]: As a part of the previous release, we bumped ancla to v0.3.4, so I'm not sure about the first bullet? Was the change already a part of the last release?

Comment (mtrinh11) [reply CHANGELOG.md:9]: The change should have been a part of the last release. I bumped ancla in pull #231 and then you bumped it in pull #232 and made a release then. I believe I didn't handle the conflicts correctly when I pulled from main and that's why that first bullet is here now.

File: common/utils.go (hunk 2 above)
Comment (mtrinh11) [reply common/utils.go:120]: Yes! Sorry, I'll get in the habit of better note tracking.

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- Bumped argus to v0.6.0. [#233](https://github.com/xmidt-org/tr1d1um/pull/233)
+- Bumped ancla to v0.3.5:
+  - Changed server log source address field. [#231](https://github.com/xmidt-org/tr1d1um/pull/231)
```
Comment (kristinapathak) [inline CHANGELOG.md:10]: I was right - this was already released in the previous version.

**Review by kristinapathak — APPROVED** [review]
